Phantom bridge

keraldi opened the pull request:
Rework.
Diagonally doesn't work properly yet, some variables are switched up which is why the template doesn't work properly.
Rest works and should work better than before.
Just cannot go speeding diagonally.

Commit:
Rework. Desyncs for no apparent reason.
Apart from that, it works.
Desyncing started after I implemented the "diagTemplate" methods.
Can definitely be optimised, some advice for that would be good (lots of duplicated code for the "templates").

Commit:
Code cleanup.
Reverted stuff that I didn't want to commit.

Commit:
Some fine tuning for speed.
playerVel is not very accurate still but this should work out for most stuff.
Might need a lot of testing.
Might still not work properly on servers.
Diagonal templates are inconsistent, need variables switched.
Will deal with that later (maybe).
In any case, this works just fine unless you're speeding diagonally.

Commit:
The occasional stuff that should not be pushed.
Meaningless whitespaces or code reformatting.

Commit:
The occasional stuff that should not be pushed.
Meaningless whitespaces or code reformatting.
Take 2.

Commit:
MERGE ME I'M AWESOME

keraldi commented:
~ finished please review ~

Commit:
removed unused import

TehNut commented on src/main/java/WayofTime/bloodmagic/item/sigil/ItemSigilPhantomBridge.java:
```diff
@@ -1,19 +1,74 @@
+        return false;
+    }
+
+    public static final String[] circle7x7 = new String[]{
```
Constants should be `CONSTANT_CASE` and should appear at the top of the class.

TehNut commented on src/main/java/WayofTime/bloodmagic/item/sigil/ItemSigilPhantomBridge.java:
```diff
@@ -1,19 +1,74 @@
-    private static final int RANGE = 3;
+    public static final Predicate<Entity> IS_PHANTOM_ACTIVE = (Entity entity) -> entity instanceof EntityPlayer && isPhantomActive((EntityPlayer) entity);
+
+    public static boolean isPhantomActive(EntityPlayer entity) {
```
Static methods should appear below all member methods.

TehNut commented on src/main/java/WayofTime/bloodmagic/item/sigil/ItemSigilPhantomBridge.java:
```diff
@@ -1,19 +1,74 @@
+            "  XXX  "
+    };
+
+    public static final String[] circle9x9 = new String[]{
```
Constants should be `CONSTANT_CASE` and should appear at the top of the class.

TehNut commented on src/main/java/WayofTime/bloodmagic/item/sigil/ItemSigilPhantomBridge.java:
```diff
@@ -1,19 +1,74 @@
+    };
+
+    //imagine you're looking into positive Z
+    public static final String[] diagXZ = new String[] {
```
Constants should be `CONSTANT_CASE` and should appear at the top of the class.

TehNut commented on src/main/java/WayofTime/bloodmagic/item/sigil/ItemSigilPhantomBridge.java:
```diff
@@ -58,32 +114,146 @@ public void onSigilUpdate(ItemStack stack, World world, EntityPlayer player, int
+            circleTemplate7x7(posX, posY, posZ, verticalOffset, world);
+        } else //anything between the first case and being slightly faster than walking
+            //walking fairly quickly on X-axis
+            if (-0.3 < playerVelZ && playerVelZ < 0.3) {
```
This reads very strangely. It should be `playerVelZ > -0.3 && playerVelZ < 0.3`. I understand you're trying to show a range, but it just makes it more confusing.

Same for all the other ranges done this way.

TehNut commented on src/main/java/WayofTime/bloodmagic/item/sigil/ItemSigilPhantomBridge.java:
```diff
@@ -58,32 +114,146 @@ public void onSigilUpdate(ItemStack stack, World world, EntityPlayer player, int
+                    rectangleTemplateNegZ(posX, posY, posZ, verticalOffset, world);
+                }
+            }
+            else //diagonal movement
```
This should have braces. It should also be on the same line as the brace above.

TehNut commented on src/main/java/WayofTime/bloodmagic/item/sigil/ItemSigilPhantomBridge.java:
```diff
@@ -58,32 +114,146 @@ public void onSigilUpdate(ItemStack stack, World world, EntityPlayer player, int
+        //Standing still, sneaking or walking with framerate drops
+        if (0 <= totalVel && totalVel < 0.1) {
+            circleTemplate7x7(posX, posY, posZ, verticalOffset, world);
+        } else //anything between the first case and being slightly faster than walking
```
This needs a brace.

TehNut commented on src/main/java/WayofTime/bloodmagic/item/sigil/ItemSigilPhantomBridge.java:
```diff
@@ -58,32 +114,146 @@ public void onSigilUpdate(ItemStack stack, World world, EntityPlayer player, int
+            else i++;
+            int j = 0;
+            for (char b: a.toCharArray()) {
+                if(inverted && !XnZ || XnZ && !inverted) j--;
```
Newlines for this ifelse

TehNut reviewed (changes requested):
Initial pass. I'm too tired to actually go over the logic right now.

I'm not a huge fan of all those 1 line helper methods.

Commit:
Refactoring, adding braces, renaming constants etc etc.

done.

keraldi commented:
Done.

Commit:
take 2

Commit:
take 2

Commit:
Removed one-liners
they grew organically because I initially created every "bridge" with a unique for loop before I removed duplicate code by passing variables/ranges
they were helpful until I found good names for the variables, now they've outlived their usefulness

keraldi commented:
Alternative explaination: I wanted more code.

Commit:
tiny bit of fine tuning

Commit:
various fluff / small fixes
more fine tuning, code reformatting, meaningful variables, for loop fix (variable mismatch), shrinking a redundant method to a call to an advanced method

TehNut commented:
My brain is too fried to try and understand this. If this doesn't work, you're toast.
